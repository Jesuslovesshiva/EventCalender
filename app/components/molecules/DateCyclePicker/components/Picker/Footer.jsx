import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {FormattedMessage} from 'react-intl'

import Button, {BUTTON_TYPES} from '@/components/atoms/Button'
import commonMessages from '@/components/common-messages'

const Footer = props => (
  <div className="flex justify-end gap-2 mx-6 pt-1.5">
    <Button onClick={props.onCancel} type={BUTTON_TYPES.OUTLINED}>
      <FormattedMessage {...commonMessages.cancel} />
    </Button>
    <Button onClick={props.onConfirm} type={BUTTON_TYPES.CONTAINED}>
      <FormattedMessage {...commonMessages.apply} />
    </Button>
  </div>
)

Footer.propTypes = {
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func
}

Footer.defaultProps = {
  onCancel: noop,
  onConfirm: noop
}

export default Footer
