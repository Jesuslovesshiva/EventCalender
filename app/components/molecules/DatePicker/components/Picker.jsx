import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {useCallback} from 'react'

import Calendar from '@/components/molecules/Calendar'

const Picker = props => {
  const onChange = useCallback(
    (...args) => {
      props.onChange(...args)
      props.close()
    },
    [props.onChange, props.close]
  )

  return (
    <Calendar
      getEventsForDate={props.getEventsForDate}
      mode="single"
      onChange={onChange}
      value={props.value}
    />
  )
}

Picker.propTypes = {
  close: PropTypes.func,
  getEventsForDate: PropTypes.func,
  onChange: PropTypes.func,
  value: PropTypes.instanceOf(Date)
}

Picker.defaultProps = {
  close: noop,
  getEventsForDate: noop,
  onChange: noop,
  value: null
}

export default Picker
