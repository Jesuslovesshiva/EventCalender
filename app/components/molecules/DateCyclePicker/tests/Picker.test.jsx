import {fireEvent} from '@testing-library/react'

import commonMessages from '@/components/common-messages'
import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'

import Picker from '../components/Picker'

const {clearDispatch, options} = configureTestStore()
const renderComponent = (props, renderFn = render) =>
  renderFn(<Picker value={new Date(2022, 2, 7)} {...props} />, options)

describe('DateCyclePicker', () => {
  describe('Picker', () => {
    beforeAll(() => {
      jest.useFakeTimers().setSystemTime(new Date(2022, 2, 7).getTime())
    })

    beforeEach(clearDispatch)

    it('should match the snapshot', () => {
      const {container} = renderComponent()
      expect(container.firstChild).toMatchSnapshot()
    })

    it('should update internal value on prop value change', () => {
      const {getByText, rerender} = renderComponent({
        value: new Date('2022-03-21')
      })
      expect(getByText('21').parentElement).toHaveClass('specific:bg-primary')
      renderComponent({value: new Date('2022-03-14')}, rerender)
      expect(getByText('21').parentElement).not.toHaveClass(
        'specific:bg-primary'
      )
      expect(getByText('14').parentElement).toHaveClass('specific:bg-primary')
    })

    describe('events', () => {
      it('should highlight whole week on day hover', () => {
        const {getByText} = renderComponent()
        const dayButton = getByText('10').parentElement
        const weekStartDayButton = getByText('7').parentElement
        const weekEndDayButton = getByText('13').parentElement
        fireEvent.mouseEnter(dayButton)
        expect(dayButton).toHaveClass('bg-gray-light')
        expect(weekStartDayButton).toHaveClass('bg-silver rounded-l')
        expect(weekEndDayButton).toHaveClass('bg-silver rounded-r')
        fireEvent.mouseLeave(dayButton)
        expect(dayButton).not.toHaveClass('bg-gray-light')
        expect(weekStartDayButton).not.toHaveClass('bg-silver rounded-l')
        expect(weekEndDayButton).not.toHaveClass('bg-silver rounded-r')
      })

      it('should call onChange prop on day click if not isConfirmable', () => {
        const onChange = jest.fn()
        const {getByText} = renderComponent({
          isConfirmable: false,
          onChange,
          value: new Date(2022, 2, 14)
        })
        const dayButton = getByText('10').parentElement
        fireEvent.click(dayButton)
        expect(onChange).toHaveBeenCalledWith(new Date(2022, 2, 7))
      })

      it('should call onChange prop on day click if isConfirmable after confirm', () => {
        const onChange = jest.fn()
        const {getByText} = renderComponent({
          isConfirmable: true,
          onChange,
          value: new Date(2022, 2, 14)
        })
        const dayButton = getByText('10').parentElement
        const confirmButton = getByText(commonMessages.apply.defaultMessage)
        fireEvent.click(dayButton)
        expect(onChange).not.toHaveBeenCalled()
        fireEvent.click(confirmButton)
        expect(onChange).toHaveBeenCalledWith(new Date(2022, 2, 7))
      })
    })
  })
})
