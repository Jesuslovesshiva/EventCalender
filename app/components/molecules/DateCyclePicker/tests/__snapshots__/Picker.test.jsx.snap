// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCyclePicker Picker should match the snapshot 1`] = `
<div
  class="rdp specific:m-0"
>
  <div
    class="rdp-months"
  >
    <div
      class="rdp-month rdp-caption_start rdp-caption_end"
    >
      <div
        class="flex items-center"
      >
        <button
          aria-label="vorheriges Monat"
          class="mr-auto p-2 text-gray-medium hover:text-gray-dark transition"
        >
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-lg"
            data-icon="arrow_back_ios"
          />
        </button>
        <div
          class="text-base w-full specific:w-32 css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="!min-h-0 !rounded-1 !transition-shadow pl-3 gap-1.5 !bg-transparent !border-gray-medium hover:!border-surface hover:!bg-silver hover:!shadow-md css-1s2u09g-control"
          >
            <div
              class="!py-0 !px-0 css-319lph-ValueContainer"
            >
              <div
                class="!m-0 !p-0 !text-gray-dark css-qc6sy-singleValue"
              >
                März
              </div>
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Monat auswählen"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-2-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <div
                aria-hidden="true"
                class="text-2xl !px-3 !py-1 !text-gray-medium css-tlfecz-indicatorContainer"
              >
                <div
                  class="fill-current w-[1em] h-[1em] inline-block"
                  data-icon="arrow_drop_down"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="text-base w-full specific:w-24 ml-1 css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="!min-h-0 !rounded-1 !transition-shadow pl-3 gap-1.5 !bg-transparent !border-gray-medium hover:!border-surface hover:!bg-silver hover:!shadow-md css-1s2u09g-control"
          >
            <div
              class="!py-0 !px-0 css-319lph-ValueContainer"
            >
              <div
                class="!m-0 !p-0 !text-gray-dark css-qc6sy-singleValue"
              >
                2022
              </div>
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Jahr auswählen"
                aria-readonly="true"
                class="css-mohuvp-dummyInput-DummyInput"
                id="react-select-3-input"
                inputmode="none"
                role="combobox"
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <div
                aria-hidden="true"
                class="text-2xl !px-3 !py-1 !text-gray-medium css-tlfecz-indicatorContainer"
              >
                <div
                  class="fill-current w-[1em] h-[1em] inline-block"
                  data-icon="arrow_drop_down"
                />
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="nächstes Monat"
          class="ml-auto p-2 text-gray-medium hover:text-gray-dark transition"
        >
          <div
            class="fill-current w-[1em] h-[1em] inline-block text-lg rotate-180"
            data-icon="arrow_back_ios"
          />
        </button>
      </div>
      <table
        aria-labelledby="1"
        class="rdp-table specific:mx-6"
        role="grid"
      >
        <thead
          class="rdp-head"
        >
          <tr
            class="rdp-head_row"
          >
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Mo
              </span>
              <span
                class="rdp-vhidden"
              >
                Montag
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Di
              </span>
              <span
                class="rdp-vhidden"
              >
                Dienstag
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Mi
              </span>
              <span
                class="rdp-vhidden"
              >
                Mittwoch
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Do
              </span>
              <span
                class="rdp-vhidden"
              >
                Donnerstag
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Fr
              </span>
              <span
                class="rdp-vhidden"
              >
                Freitag
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                Sa
              </span>
              <span
                class="rdp-vhidden"
              >
                Samstag
              </span>
            </th>
            <th
              class="rdp-head_cell specific:w-10 specific:h-7 specific:text-[11px] specific:leading-[13px]"
              scope="col"
            >
              <span
                aria-hidden="true"
              >
                So
              </span>
              <span
                class="rdp-vhidden"
              >
                Sonntag
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="rdp-tbody"
        >
          <tr
            class="rdp-row"
          >
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200 text-gray-medium"
                date="Mon Feb 28 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  28
                </span>
                <span
                  class="rdp-vhidden"
                >
                  28. Februar (Montag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Tue Mar 01 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  1
                </span>
                <span
                  class="rdp-vhidden"
                >
                  1. März (Dienstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Wed Mar 02 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  2
                </span>
                <span
                  class="rdp-vhidden"
                >
                  2. März (Mittwoch)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Thu Mar 03 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  3
                </span>
                <span
                  class="rdp-vhidden"
                >
                  3. März (Donnerstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Fri Mar 04 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  4
                </span>
                <span
                  class="rdp-vhidden"
                >
                  4. März (Freitag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sat Mar 05 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  5
                </span>
                <span
                  class="rdp-vhidden"
                >
                  5. März (Samstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sun Mar 06 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  6
                </span>
                <span
                  class="rdp-vhidden"
                >
                  6. März (Sonntag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
          </tr>
          <tr
            class="rdp-row"
          >
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 font-bold specific:text-white specific:bg-primary rounded-l"
                date="Mon Mar 07 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="0"
              >
                <span
                  aria-hidden="true"
                >
                  7
                </span>
                <span
                  class="rdp-vhidden"
                >
                  7. März (Montag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary"
                date="Tue Mar 08 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  8
                </span>
                <span
                  class="rdp-vhidden"
                >
                  8. März (Dienstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary"
                date="Wed Mar 09 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  9
                </span>
                <span
                  class="rdp-vhidden"
                >
                  9. März (Mittwoch)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary"
                date="Thu Mar 10 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  10
                </span>
                <span
                  class="rdp-vhidden"
                >
                  10. März (Donnerstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary"
                date="Fri Mar 11 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  11
                </span>
                <span
                  class="rdp-vhidden"
                >
                  11. März (Freitag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary"
                date="Sat Mar 12 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  12
                </span>
                <span
                  class="rdp-vhidden"
                >
                  12. März (Samstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                aria-pressed="true"
                class="w-full h-8 transition duration-200 specific:text-white specific:bg-primary rounded-r"
                date="Sun Mar 13 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  13
                </span>
                <span
                  class="rdp-vhidden"
                >
                  13. März (Sonntag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
          </tr>
          <tr
            class="rdp-row"
          >
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Mon Mar 14 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  14
                </span>
                <span
                  class="rdp-vhidden"
                >
                  14. März (Montag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Tue Mar 15 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  15
                </span>
                <span
                  class="rdp-vhidden"
                >
                  15. März (Dienstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Wed Mar 16 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  16
                </span>
                <span
                  class="rdp-vhidden"
                >
                  16. März (Mittwoch)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Thu Mar 17 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  17
                </span>
                <span
                  class="rdp-vhidden"
                >
                  17. März (Donnerstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Fri Mar 18 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  18
                </span>
                <span
                  class="rdp-vhidden"
                >
                  18. März (Freitag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sat Mar 19 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  19
                </span>
                <span
                  class="rdp-vhidden"
                >
                  19. März (Samstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sun Mar 20 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  20
                </span>
                <span
                  class="rdp-vhidden"
                >
                  20. März (Sonntag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
          </tr>
          <tr
            class="rdp-row"
          >
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Mon Mar 21 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  21
                </span>
                <span
                  class="rdp-vhidden"
                >
                  21. März (Montag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Tue Mar 22 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  22
                </span>
                <span
                  class="rdp-vhidden"
                >
                  22. März (Dienstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Wed Mar 23 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  23
                </span>
                <span
                  class="rdp-vhidden"
                >
                  23. März (Mittwoch)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Thu Mar 24 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  24
                </span>
                <span
                  class="rdp-vhidden"
                >
                  24. März (Donnerstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Fri Mar 25 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  25
                </span>
                <span
                  class="rdp-vhidden"
                >
                  25. März (Freitag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sat Mar 26 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  26
                </span>
                <span
                  class="rdp-vhidden"
                >
                  26. März (Samstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Sun Mar 27 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  27
                </span>
                <span
                  class="rdp-vhidden"
                >
                  27. März (Sonntag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
          </tr>
          <tr
            class="rdp-row"
          >
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Mon Mar 28 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  28
                </span>
                <span
                  class="rdp-vhidden"
                >
                  28. März (Montag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Tue Mar 29 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  29
                </span>
                <span
                  class="rdp-vhidden"
                >
                  29. März (Dienstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Wed Mar 30 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  30
                </span>
                <span
                  class="rdp-vhidden"
                >
                  30. März (Mittwoch)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200"
                date="Thu Mar 31 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  31
                </span>
                <span
                  class="rdp-vhidden"
                >
                  31. März (Donnerstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200 text-gray-medium"
                date="Fri Apr 01 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  1
                </span>
                <span
                  class="rdp-vhidden"
                >
                  1. April (Freitag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200 text-gray-medium"
                date="Sat Apr 02 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  2
                </span>
                <span
                  class="rdp-vhidden"
                >
                  2. April (Samstag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
            <td
              class="rdp-cell specific:w-10 specific:h-8 specific:text-[18px] specific:leading-[21px] specific:pb-2"
            >
              <button
                class="w-full h-8 transition duration-200 text-gray-medium"
                date="Sun Apr 03 2022 00:00:00 GMT+0000 (Coordinated Universal Time)"
                tabindex="-1"
              >
                <span
                  aria-hidden="true"
                >
                  3
                </span>
                <span
                  class="rdp-vhidden"
                >
                  3. April (Sonntag)
                </span>
                <div
                  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden mt-px w-[38px]"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
